app/models/claims/claim.rb

```diff
@@ -56,6 +56,22 @@ class Claims::Claim < ApplicationRecord
   delegate :users, to: :school, prefix: true
   delegate :full_name, to: :submitted_by, prefix: true, allow_nil: true
 
+  def invalid_mentor_training_hours?
+    mentor_trainings.each do |mentor_training|
+      remaining_training_hours = Claims::Mentor::CalculateRemainingMentorTrainingHoursForProvider.call(
+        mentor: mentor_training.mentor,
+        provider:,
+        claim: self,
+      )
+
+      if mentor_training.hours_completed > remaining_training_hours
+        return true
+      end
+    end
+
+    false
+  end
```

Comment:
Could this potentially be:

```ruby
class Claims::Claim
  validates_associated :mentor_trainings
end
```

And then we do a validation before the submit call?

```ruby
class Claims::Schools::ClaimsController
  def submit
    if @claim.valid?
      # perform Submit
    else
      redirect_to rejected...
    end
  end
end
```

Reply:
Hmm, I think I can see a potential issue. When creating an `internal_draft`, it _might_ be invalid.

My worry is without a validation, we may be letting invalid data through.

Reply:
I now recall that we don't actually validate `mentor_trainings` as records that effectively either. It's validated within the form object, but we need to validate the claim as a whole too.

Reply:
Addressed this, kept it as a method but changed the query